src/renderer/atlas/BackendD2D.cpp

```diff
@@ -486,6 +493,27 @@ void BackendD2D::_drawSelection(const RenderingPayload& p)
     }
 }
 
+void BackendD2D::_debugShowDirty(RenderingPayload& p)
+{
+    _presentRects[_presentRectsPos] = p.dirtyRectInPx;
+    _presentRectsPos = (_presentRectsPos + 1) % std::size(_presentRects);
+
+    for (size_t i = 0; i < std::size(_presentRects); ++i)
+    {
+        if (const auto& rect = _presentRects[i])
+        {
+            const D2D1_RECT_F rectF{
+                static_cast<f32>(rect.left),
+                static_cast<f32>(rect.top),
+                static_cast<f32>(rect.right),
+                static_cast<f32>(rect.bottom),
+            };
+            const auto color = 0x1f000000 | colorbrewer::pastel1[i];

```

Comment:
how does this not index past the end of the pastel palette?

Reply:
The size of the `_presentRects` array matches the one of the pastel colors and so it works out fine.

BTW I‘m still not happy with that particular code and I might be looking for a better alternative if time permits.